File: dataset_reduzido/codenet/Java/Code/s518927375.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        long k = sc.nextLong();
        long a = sc.nextLong();
        long b = sc.nextLong();

        if (b <= a + 2) {
            System.out.println(1 + k);
            return;
        }

        long x = k - (a - 1);
        long ans = a + (x / 2) * (b - a) + (x % 2);

        System.out.println(ans);
    }
}

File: output/Llama-2-100/codenet/Java/Python/s518927375.py
def main():
    sc = input()

    k = sc.nextLong()
    a = sc.nextLong()
    b = sc.nextLong()

    if b <= a + 2:
        print(1 + k)
        return

    x = k - (a - 1)
    ans = a + (x // 2) * (b - a) + (x % 2)

    print(ans)

